misc/formatter/modules/formatter-cli/src/main/java/org/ballerinalang/formatter/cli/FormatUtil.java

```diff
@@ -0,0 +1,358 @@
+            }
+        }
+
+        return formattedFiles;
+    }
+
+    /**
+     * Get prepared compiler context.
+     *
+     * @param sourceRootPath ballerina compilable source root path
+     * @return {@link CompilerContext} compiler context
+     */
+    private static CompilerContext getCompilerContext(Path sourceRootPath) {
+        CompilerPhase compilerPhase = CompilerPhase.DEFINE;
+        CompilerContext context = new CompilerContext();
+        CompilerOptions options = CompilerOptions.getInstance(context);
+        options.put(PROJECT_DIR, sourceRootPath.toString());
+        options.put(OFFLINE, Boolean.toString(false));
+        options.put(COMPILER_PHASE, compilerPhase.toString());
+        options.put(SKIP_TESTS, Boolean.toString(false));
+        options.put(TEST_ENABLED, "true");
+        options.put(LOCK_ENABLED, Boolean.toString(false));
+        options.put(EXPERIMENTAL_FEATURES_ENABLED, Boolean.toString(true));
+        options.put(PRESERVE_WHITESPACE, Boolean.toString(true));
+        options.put(NEW_PARSER_ENABLED, Boolean.toString(false));
+
+        return context;
+    }
+
+    /**
+     * Check whether the given module name exists.
+     *
+     * @param module      module name
+     * @param projectRoot path of the ballerina project root
+     * @return {@link Boolean} true or false
+     */
+    private static boolean isModuleExist(String module, Path projectRoot) {
+        Path modulePath;
+        if (module.startsWith("src/")) {
+            modulePath = projectRoot.resolve(module);
+        } else {
+            modulePath = projectRoot.resolve("src").resolve(module);
+        }
+
+        return modulePath.toFile().isDirectory();
+    }
+
+    /**
+     * Check whether the given path isn't a source root of a ballerina project.
+     *
+     * @param path - path where the command is executed from
+     * @return {@link boolean} true or false
+     */
+    private static boolean notABallerinaProject(Path path) {
+        Path cachePath = path.resolve("Ballerina.toml");
+        return !cachePath.toFile().exists();
+    }
+
+    /**
+     * Write content to a file.
+     *
+     * @param filePath - path of the file to add the content
+     * @param content  - content to be added to the file
+     * @throws IOException - throws and IO exception
+     */
+    private static void writeFile(String filePath, String content) throws IOException {
+        OutputStreamWriter fileWriter = null;
+        try {
+            try (FileOutputStream fileStream = new FileOutputStream(new File(filePath))) {
+                fileWriter = new OutputStreamWriter(fileStream, StandardCharsets.UTF_8);
+            }
+            fileWriter.write(content);
+        } finally {
+            if (fileWriter != null) {
+                fileWriter.close();
+            }
+        }
+    }
+
+    private static boolean doChangesAvailable(String originalSource, String formattedSource) {
```

Comment:
Does `areChangesAvailable` makes sense instead of `doChangesAvailable` ?

Reply:
Done